### PR TITLE
Improve documentation for backfilling rooms

### PR DESCRIPTION
**Pull request by turt2live**
Rendered: see 'docs' commit status

----

There's not a whole lot to improve here - most of the changes are about reorganization and minor clarifications.

Most of my time was spent verifying that the endpoint does what it says it does: namely that it looks up previous events for a given set.

Fixes https://github.com/matrix-org/matrix-doc/issues/1385

### CODE REVIEW

File: api/server-server/backfill.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                description: The maximum number of events to retrieve. Defaults to 10.
+                example: 10
+              min_depth:
+                type: integer
+                description: The minimum depth of events to retrieve. Defaults to 0.
+                example: 0
+              earliest_events:
+                type: array
+                description: |-
+                  The events the sender is missing. The response will only contain events
```
Comment (erikjohnston) [inline api/server-server/backfill.yaml]: We have the events in `earliest_events`, no? (Otherwise how do we know about them?

Comment (turt2live) [reply api/server-server/backfill.yaml]: These are the `prev_events` to the `latest_events` that the sender is missing, so it only knows the event IDs.

Comment (erikjohnston) [reply api/server-server/backfill.yaml]: https://github.com/matrix-org/synapse/blob/master/synapse/handlers/federation.py#L344 suggests that they are latest events the sender already has?

Comment (turt2live) [reply api/server-server/backfill.yaml]: Looks like it, although I'm suitably confused as to how this endpoint works again. For the sake of understanding: `earliest_events` are the edges the server has and `latest_events` are the events the server wants? I'm not totally understanding what the receiver is supposed to do with this information.

Comment (erikjohnston) [reply api/server-server/backfill.yaml]: Its to get all the missing events between `earliest_events` (which the server has), and `latest_events` (which the server has just found out about, but doesn't have the prev events for).

**Pull request merged**